models/articles.model.js

```diff
@@ -12,28 +12,43 @@ exports.fetchArticleById = (id) => {
         })
 }
 
-exports.fetchArticles = (topic) => {
-    let queryStr = `
+exports.fetchArticles = (query, queryKeys) => {
+    const topics = []
+    const sqlValue = []
+    const validQuery = ['topic']
+    const allValid = queryKeys.every((query)=>validQuery.includes(query))
+
+    if (!allValid) return Promise.reject({status: 400, message: 'Invalid query'})
+
+    let sqlStr = `
     SELECT articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count 
     FROM articles LEFT JOIN comments
     ON articles.article_id = comments.article_id
     `
-    const queryValue = []
 
-    if (topic) {
-        queryStr += `WHERE articles.topic = $1`
-        queryValue.push(topic)
+    if (query.topic) {
+        sqlStr += `WHERE articles.topic = $1`
+        sqlValue.push(query.topic)
     }
 
-    queryStr += `
+    sqlStr += `
     GROUP BY articles.article_id
     ORDER BY articles.created_at DESC;
     `
-
-    return db.query(queryStr, queryValue)
-        .then(({ rows }) => {
-            return rows.length === 0 ? Promise.reject({ status: 404, message: 'article not found' }) : rows;
-        })
+    return db.query(`SELECT slug FROM topics;`)
+    .then(({rows})=>{
```

Comment:
I would separate out this query and call it within this function :) - We dont want to have multiple queries in here like this!